src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -369,10 +369,10 @@ subroutine Infiltration(bounds, num_hydrologyc, filter_hydrologyc, num_urbanc, f
              else
                 qinmax=(1._r8 - fsat(c)) * minval(10._r8**(-e_ice*(icefrac(c,1:3)))*hksat(c,1:3))
              end if
-             qflx_infl_excess(c) = max(0._r8,qflx_in_soil(c) -  (1.0_r8 - frac_h2osfc(c))*qinmax)
+             qflx_infl_excess(c) = max(0._r8, qflx_in_soil(c) - qinmax) * (1.0_r8 - frac_h2osfc(c)) ! qflx_infl_excess only over the soil
```

Comment:
**Comment from @martynpclark Aug 17, 2017**

Code modified because using the spatially constant forcing. The infiltration excess runoff only occurs over the portion of the grid cell that is not covered with ponded water.